core/src/main/java/org/infinispan/topology/ClusterTopologyManagerImpl.java

```diff
@@ -448,9 +448,18 @@ private void recoverClusterStatus(int newViewId, final boolean isMergeView, fina
                         new CacheTopologyFilterReuser());
             break;
          } catch (ExecutionException e) {
-            if (i == 0 || !(e.getCause() instanceof TimeoutException))
-               throw e;
-            log.debug("Timed out waiting for cluster status responses, trying again");
+            if (i != 0) {
+               if (e.getCause() instanceof TimeoutException) {
+                  log.debug("Timed out waiting for cluster status responses, trying again");
+               } else if (e.getCause() instanceof SuspectException) {
+                  if (transport.getMembers().containsAll(clusterMembers)) {
+                     log.debug("Received a CacheNotFoundResponse from one of the members, trying again");
+                     Thread.sleep(getGlobalTimeout() / CLUSTER_RECOVERY_ATTEMPTS / 2);
```

Comment:
@rvansa The problem I was having is that with FORK, a node would become a member in the JGroups view before having a `CommandAwareRpcDispatcher`, and if we didn't wait at all we'd be using up all the retries in a few milliseconds (see `ConcurrentStartForkChannelTest`). Perhaps we could avoid the wait if we had an explicit "cache manager join" command to tell the coordinator which nodes are really running Infinispan.